src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs

```diff
@@ -749,44 +750,52 @@ protected virtual void WriteLROMethodBody(MgmtClientOperation clientOperation, D
             }
         }
 
-        protected virtual void WriteLROMethodBranch(CSharpType lroObjectType, MgmtRestOperation operation, IEnumerable<ParameterMapping> parameterMapping, bool async)
+        protected virtual void WriteLROMethodBranch(MgmtRestOperation operation, IEnumerable<ParameterMapping> parameterMapping, bool async)
         {
             _writer.Append($"var response = {GetAwait(async)} ");
             _writer.Append($"{GetRestClientName(operation)}.{CreateMethodName(operation.Method.Name, async)}(");
             WriteArguments(_writer, parameterMapping);
             _writer.Line($"cancellationToken){GetConfigureAwait(async)};");
 
-            WriteLROResponse(lroObjectType, GetDiagnosticName(operation), PipelineProperty, operation, parameterMapping, async);
+            WriteLROResponse(GetDiagnosticName(operation), PipelineProperty, operation, parameterMapping, async);
         }
 
-        protected virtual void WriteLROResponse(CSharpType lroObjectType, string diagnosticsVariableName, string pipelineVariableName, MgmtRestOperation operation, IEnumerable<ParameterMapping> parameterMapping, bool async)
+        protected virtual void WriteLROResponse(string diagnosticsVariableName, string pipelineVariableName, MgmtRestOperation operation, IEnumerable<ParameterMapping> parameterMapping, bool async)
         {
-            _writer.Append($"var operation = new {lroObjectType}(");
-
-            if (operation.Operation.IsLongRunning)
+            _writer.Append($"var operation = new {LibraryArmOperation}");
+            if (operation.ReturnType.Arguments.Length > 0)
+            {
+                _writer.Append($"<{operation.ReturnType.Arguments[0]}>");
+            }
```

Comment:
Do you think it makes sense if we could put this type `LibraryArmOperation<operation.ReturnType.Arguments[0]>` as a property of `MgmtRestOperation`?